### PR TITLE
Feature/issue16/openwhisk brokerage

### PR DESCRIPTION
**Pull request by Zee-Khan**

### CODE REVIEW

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/openwhisk/OpenWhiskServiceProvider.groovy [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import static com.swisscom.cloud.sb.broker.util.ServiceDetailKey.OPENWHISK_EXECUTION_URL
+import static com.swisscom.cloud.sb.broker.util.ServiceDetailKey.OPENWHISK_KEY
+import static com.swisscom.cloud.sb.broker.util.ServiceDetailKey.OPENWHISK_NAMESPACE
+import static com.swisscom.cloud.sb.broker.util.ServiceDetailKey.OPENWHISK_SUBJECT
+import static com.swisscom.cloud.sb.broker.util.ServiceDetailKey.OPENWHISK_UUID
+
+@Component
+@CompileStatic
+@Slf4j
+class OpenWhiskServiceProvider implements ServiceProvider{
```
Comment (mcelep) [inline broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/openwhisk/OpenWhiskServiceProvider.groovy]: This class should implement the ServiceUsageProvider interface

File: broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/OpenwhiskFunctionalSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        def credentials = serviceLifeCycler.getCredentials()
+        println("Credentials: ${credentials}")
+        then:
+        noExceptionThrown()
+    }
+
+    def "create and execute action"() {
+        when:
+        def credentials = serviceLifeCycler.getCredentials()
+        restTemplate = restTemplatefactory.buildWithSSLValidationDisabledAndBasicAuthentication(credentials.get("uuid"), credentials.get("key"))
```
Comment (Zee-Khan) [inline broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/OpenwhiskFunctionalSpec.groovy]: Done.

File: broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/OpenwhiskFunctionalSpec.groovy [hunk 3]
```diff
@@ -36,26 +36,51 @@ class OpenwhiskFunctionalSpec extends BaseFunctionalSpec {
...
         restTemplate = restTemplatefactory.buildWithSSLValidationDisabledAndBasicAuthentication(credentials.get("uuid"), credentials.get("key"))
-        String packagePayload = """{"version": "0.0.1", "publish": true}"""
         HttpHeaders headers = new HttpHeaders()
         headers.setContentType(MediaType.APPLICATION_JSON)
-        HttpEntity<String> packageEntity = new HttpEntity<String>(packagePayload, headers)
+
+        createPackage(credentials, headers)
+        createAction(credentials, headers)
+        executeAction(credentials)
+        serviceLifeCycler.getUsage()
```
Comment (mcelep) [inline broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/OpenwhiskFunctionalSpec.groovy]: serviceBrokerClient.getUsage() can be used instead and all the changes introduced for calling serviceLifeCycler.getUsage() can be reverted back.

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/controller/CFExtensionsController.groovy [hunk 4]
```diff
@@ -37,7 +37,7 @@ class CFExtensionsController extends BaseController {
     def usage(
             @PathVariable('service_instance') String serviceInstanceId,
             @RequestParam(value = 'end_date', required = false) String enddate) {
-        return serviceUsageLookup.usage(findServiceInstance(serviceInstanceId), parseEnddate(enddate))
+        return serviceUsageLookup.usage(super.getAndCheckServiceInstance(serviceInstanceId), parseEnddate(enddate))
```
Comment (mcelep) [inline broker/src/main/groovy/com/swisscom/cloud/sb/broker/controller/CFExtensionsController.groovy]: using 'super.' is not required in this case.

**Pull request merged**